### ISSUE TITLE
goto parsing fails with  empty target

### ISSUE THREAD

**Issue by jrobeson** [microsoft/tolerant-php-parser#154]
Symfony's route generator generates a url matcher on disk like so:

```php
<?php
class UrlMatcher {
    public function match($path)
    {
        if ('/path' === $pathinfo) {
            if ('POST' !== $canonicalMethod) {
                $allow[] = 'POST';
                goto not_this_path;
            }
            return array ('_controller' => 'method',  '_route' => 'this_path');
        }
        not_this_path:
        // other routes with their own goto not_that_path
    }
}
```
PHP seems to allow empty targets, since it runs the code just fine.  This is a regression compared to the previous version of the language server.

**Comment by jens1o** [comment]
When you add any expression below it, it passes: 
![image](https://user-images.githubusercontent.com/11234139/27251997-6ec493a4-5353-11e7-89f6-1f143de7ebb7.png)

**Comment by jrobeson** [comment]
I know, but the current behaviour is still wrong vs PHP itself.

**Comment by roblourens** [comment]
Hm, so in this case, the next token after the `label:` is always a close brace or EOF, right (besides whitespace, comment, etc)?

**Comment by roblourens** [comment]
There's also this case:

```
goto label;
if (false)
    label:
else
    echo "else\n";
```

This seems hard to parse. Will think about it.

**Comment by jrobeson** [comment]
@roblourens : as far as I can tell yes, but my router usage is so minimal that I'm not sure what other corner cases there might be.

**Comment by jrobeson** [comment]
shouldn't this also be tagged as lang server regression? I don't think it errored out on this.
